src/config/reader.rs

```diff
@@ -247,7 +246,7 @@ impl ConfigReader {
             }
         }
 
-        descriptors.insert(proto_path, parent_proto);
+        descriptors.insert(String::new(), parent_proto);
```

Comment:
that seems odd to put into hashmap the entry with empty key. If it's intentional how this should work?

Reply:
we never need the key outside the function, the key just used to avoid remove any cycle dependencies or adding same file several times

Reply:
in that case I suggest to keep such data itself that function only and accept/return only the data that is required i.e. return the list of descriptors instead of map to prevent any misuses

Reply:
I removed hashmap from arg and made the function return vector